File: abstract-factory/go/src/vehicle/Van.go
package vehicle

import (
  "fmt"
)

// 具体产品实现类
type Van struct {
  Name string
}

func (v Van) GetName() string {
  return v.Name
}

func (v *Van) Run() {
  fmt.Println("Van::Run() [Name=" + v.Name + "]")
}

File: abstract-factory/c/src/vehicle/van.c
#include "../func.h"

/* Van对象函数实现 */
void van_run(Vehicle *vehicle)
{
  printf("\r\n van_run() [name=%s]", vehicle->name);
}

/* 创建对象构造函数 */
Van *van_constructor(char *name)
{
  Van *van = (Van *)malloc(sizeof(Van));
  strncpy(van->name, name, 50);
  van->run = &van_run;
  return van;
}